modules/services/imapnotify.nix

```diff
@@ -9,49 +9,58 @@ let
   safeName = lib.replaceChars ["@" ":" "\\" "[" "]"] ["-" "-" "-" "" ""];
 
   imapnotifyAccounts =
-    filter (a: a.imapnotify.enable) (attrValues config.accounts.email.accounts);
-
-  genAccountUnit = account: {
-    name = "imapnotify@${safeName account.name}";
-    value = {
-      Unit = {
-        Description = "Execute scripts on IMAP mailbox changes (new/deleted/updated messages) using IDLE for %i";
-        After = [ "gpg-agent.service" ];
-      };
+    filter (a: a.imapnotify.enable)
+    (attrValues config.accounts.email.accounts);
+
+  genAccountUnit = account:
+    let
+      name = safeName account.name;
+    in
+      {
+        name = "imapnotify-${name}";
+        value = {
+          Unit = {
+            Description = "ImapNotify for ${name}";
+          };
 
-      Service = {
-        ExecStart = "${pkgs.imapnotify}/bin/imapnotify -c ${genAccountConfig account}";
-      };
+          Service = {
+            ExecStart = "${pkgs.imapnotify}/bin/imapnotify -c ${genAccountConfig account}";
+          };
 
-      Install = {
-        WantedBy = [ "default.target" ];
+          Install = {
+            WantedBy = [ "default.target" ];
+          };
+        };
       };
-    };
-  };
 
   genAccountConfig = account:
     pkgs.writeText "imapnotify-${safeName account.name}-config.js" (
-        let
-          port = if account.imap.port != null
-                   then account.imap.port
-                   else if account.imap.tls.enable then 993 else 143;
-        in ''
+      let
+        port =
+          if account.imap.port != null then account.imap.port
+          else if account.imap.tls.enable then 993
+          else 143;
+
+        toJSON = builtins.toJSON;
+      in
+        ''
           var child_process = require('child_process');
 
           function getStdout(cmd) {
               var stdout = child_process.execSync(cmd);
               return stdout.toString().trim();
           }
 
-          exports.host = "${account.imap.host}"
-          exports.port = ${toString port};
-          exports.tls = ${builtins.toJSON account.imap.tls.enable};
-          exports.username = "${account.userName}";
-          exports.password = getStdout("${toString account.passwordCommand}");
-          exports.onNotify = ${builtins.toJSON account.imapnotify.onNotify};
-          exports.onNotifyPost = ${builtins.toJSON account.imapnotify.onNotifyPost};
-          exports.boxes = ${builtins.toJSON account.imapnotify.boxes};
-        '');
+          exports.host = ${toJSON account.imap.host}
+          exports.port = ${toJSON port};
+          exports.tls = ${toJSON account.imap.tls.enable};
+          exports.username = ${toJSON account.userName};
+          exports.password = getStdout(${toJSON account.passwordCommand});
```

Comment:
This doesn't work, because it produces something like
`getStdout(["cmd", "arg0", "arg1"]);`
when according to https://nodejs.org/api/child_process.html#child_process_child_process_execsync_command_options it needs to take a string